pyteal/ast/jsonref.py

```diff
@@ -0,0 +1,107 @@
+from typing import TYPE_CHECKING
+from enum import Enum
+
+from pyteal.types import TealType, require_type
+from pyteal.errors import verifyFieldVersion, verifyTealVersion
+from pyteal.ir import TealOp, Op, TealBlock
+from pyteal.ast.expr import Expr
+from pyteal.ast.leafexpr import LeafExpr
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+class JsonRefType(Enum):
+    # fmt: off
+    #              id |    name   |      type      | min version
+    string = (0, "JSONString", TealType.bytes,  7)
+    uint64 = (1, "JSONUint64", TealType.uint64, 7)
+    object = (2, "JSONObject", TealType.bytes,  7)
+    # fmt: on
+
+    def __init__(self, id: int, name: str, type: TealType, min_version: int) -> None:
+        self.id = id
+        self.arg_name = name
+        self.ret_type = type
+        self.min_version = min_version
+
+    def type_of(self) -> TealType:
+        return self.ret_type
+
+
+JsonRefType.__module__ = "pyteal"
+
+
+class JsonRef(LeafExpr):
```

Comment:
Prefer `Expr` over `LeafExpr`

Reply:
@jasonpaulos I attempted to address in https://github.com/algorand/pyteal/pull/452.  Can you have a look and resolve if the PR captures it?